src/asmcnc/skavaUI/screen_go.py

```diff
@@ -691,13 +688,33 @@ def start_or_pause_button_press(self):
 
                 self.m.s.is_ready_to_assess_spindle_for_shutdown = True # allow spindle overload assessment to resume
         else:
-            self._start_running_job()
-            self.jd.job_start_time = time.time()
+            has_health_check_run = self.m.has_spindle_health_check_passed() or self.m.has_spindle_health_check_failed()
+
+            self.m._grbl_soft_reset()
+
+            if self.m.is_spindle_health_check_active() \
+                    and not has_health_check_run and self.m.stylus_router_choice == 'router':
+                self.run_spindle_health_check(start_after_pass=True)
+            else:
+                self._start_running_job()
+
+                if not self.m.has_spindle_health_check_passed():
```

Comment:
doesn't this need to be if self.m.has_spindle_health_check_failed(): ?

Reply:
no because they can fail and run